src/main/java/com/siit/hospital_manager/controller/AppointmentController.java

```diff
@@ -21,27 +27,138 @@ public class AppointmentController {
 
     private final AppointmentService appointmentService;
     private final DoctorService doctorService;
+    private final PatientService patientService;
+    private final SpecialtyRepository specialtyRepository;
+    private final DiagnosisService diagnosisService;
+    private final ProcedureService procedureService;
+    private final MedicationService medicationService;
 
-    @GetMapping("/findAllByPatient")
+    private final EmailSender emailSender;
+
+    @GetMapping("/viewAll")
     public String findAllByPatient(Model model, Principal principal) {
         List<AppointmentDto> appointments = appointmentService.findAllByUserName(principal.getName());
         model.addAttribute("appointments", appointments);
-
         return "appointment/viewAll";
     }
 
+    @GetMapping("/appointmentDetails/{apId}")
+    public String findAllDetailsByPatient(@PathVariable Integer apId,Model model) {
+        model.addAttribute("appointment", appointmentService.findById(apId));
+        model.addAttribute("appointmentDiagnoses", appointmentService.findById(apId).getDiagnoses());
+        model.addAttribute("appointmentProcedures", appointmentService.findById(apId).getProcedures());
+        model.addAttribute("appointmentMedications", appointmentService.findById(apId).getMedications());
+        return "appointment/appointmentDetails/appointmentsDetails";
+    }
+
+
+
     @DeleteMapping("/{id}")
     @ResponseStatus(HttpStatus.ACCEPTED)
     public void deleteAppointmentById(Model model, @PathVariable Integer id, Principal principal){
          appointmentService.deleteAppointmentByIdAndPatient(id, principal.getName());
     }
 
-    @GetMapping("/create")
-    public String create(Model model, Principal principal) {
-        model.addAttribute("doctors", doctorService.findAll());
+    @GetMapping("/create/{specialtyId}")
+    public String showCreateSpecialtyForm(@PathVariable Integer specialtyId, Model model) {
+        Specialty specialty = specialtyRepository.findById(specialtyId)
+                .orElseThrow(() -> new EntityNotFoundException("Specialty not found"));
```

Comment:
this logic is specific to a service, do you have a global handler method to return a view for EntityNotFoundException?